### PR TITLE
feat: add R128 based linear normalization

### PR DESCRIPTION
**Pull request by KillerX**

### CODE REVIEW

File: utils/execute_command.go [hunk 1]
```diff
@@ -47,3 +49,44 @@ func ExecuteCmd(cmd *exec.Cmd, outputCallback func(string)) (string, error) {
 
 	return result, err
 }
+
+// ExecuteAnalysisCmd executes the cmd and returns through outputCallback line-by-line and retrutning only
+// the part that is a valid JSON string
+func ExecuteAnalysisCmd(cmd *exec.Cmd, outputCallback func(string)) (string, error) {
```
Comment (fredrikved) [inline utils/execute_command.go:55]: This analysis can take time as well (for meetings, etc). Wouldn't piping progress to another pipe be better?

File: activities/normalize.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	"go.temporal.io/sdk/activity"
+)
+
+type AnalyzeEBUR128Params struct {
+	FilePath       string
+	TargetLoudness float64
+}
+
+type AnalyzeEBUR128Result struct {
+	IntegratedLoudnes   float64
```
Comment (andreasgangso) [inline activities/normalize.go:18]: IntegratedLoudness

File: activities/normalize.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	TruePeak            float64
+	LoudnesRange        float64
```
Comment (andreasgangso) [inline activities/normalize.go:20]: LoudnessRange

File: services/ffmpeg/ebur128.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strconv"
+
+	"github.com/bcc-code/bccm-flows/utils"
+)
+
+type loudnormResult struct {
+	InputIntegratedLoudnes string `json:"input_i"`
```
Comment (andreasgangso) [inline services/ffmpeg/ebur128.go:13]: Loudness

File: activities/normalize.go [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+		Path:            input.InFilePath,
+		DestinationPath: input.OutFilePath,
+	}, input.Adjustment, progressCallback)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &LinearAdjustAudioResult{
+		OutFilePath: res.OutputPath,
```
Comment (andreasgangso) [inline activities/normalize.go:85]: Maybe reuse the LinearNormalizeAudio result type (AudioResult)
to decrease the amount of types and mappings

File: activities/normalize.go [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+	"github.com/bcc-code/bccm-flows/services/transcode"
+	"go.temporal.io/sdk/activity"
+)
+
+type AnalyzeEBUR128Params struct {
+	FilePath       string
+	TargetLoudness float64
+}
+
+type AnalyzeEBUR128Result struct {
```
Comment (andreasgangso) [inline activities/normalize.go:17]: Maybe reuse this result type (move it to common?), because it's duplicated three times now

**Review by andreasgangso — CHANGES_REQUESTED** [review]

File: services/transcode/linear_normalize.go [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+	outputPath = outputPath[:len(outputPath)-len(filepath.Ext(outputPath))] + "_normalized" + filepath.Ext(outputPath)
+
+	params := []string{
+		"-i", input.Path,
+		"-c:v", "copy",
+		"-af", fmt.Sprintf("volume=%.2fdB", adjustment),
+		outputPath,
+	}
+
+	//replace output extension to .aac
```
Comment (fredrikved) [inline services/transcode/linear_normalize.go:23]: inaccurate comment?

**Review by fredrikved — APPROVED** [review]

File: services/transcode/linear_normalize.go [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/bcc-code/bccm-flows/common"
+	"github.com/bcc-code/bccm-flows/services/ffmpeg"
+)
+
+func LinearNormalizeAudio(input common.AudioInput, adjustment float64, cb ffmpeg.ProgressCallback) (*common.AudioResult, error) {
```
Comment (andreasgangso) [inline services/transcode/linear_normalize.go:12]: denne funksjonen gjør ikke normalization så LinearAdjustAudio er bedre

**Review by andreasgangso — CHANGES_REQUESTED** [review]

**Review by andreasgangso — APPROVED** [review]

**Pull request merged**